File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    count = 0
    for i in range(len(s)):
        if (s[i] == "("):
            count += 1
        else:
            count -= 1
        if count < 0:
            answer = False
            break
    if (count != 0):
        answer = False
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        
        Stack<Character> stack = new Stack();
        
        for (char c : s.toCharArray()){
            if (c == '('){
                
                stack.push(c);
                
            }
            else{
                if (!stack.isEmpty() && stack.peek() == '(')
                    stack.pop();
                else{
                    answer = false;
                }
            }
        }
        
        if (!stack.isEmpty())
            answer = false;

        return answer;
    }
}